File: SOLUTIONS/reverse-word.scala
object ReverseWord extends App {
  def reverseWord(): String = {
    print("Enter a sentence to reverse words order: ")
    val sentence = scala.io.StdIn.readLine()
    println()
    val words = sentence.split(" ")
    val reversed_words = words.reverse
    reversed_words.mkString(" ")
  }
  
  println(reverseWord())
}

File: SOLUTIONS/reverse_word.hs
-- Code developed by Quino Terrasa (@espetro) for the
-- 2018 Hacktoberfest BCN

-- run 'ghci reverse_words.hs' then type 'main' to get it working
main = do
    putStrLn "Enter a sentence"
    str <- getLine
    let reversed = (concatMap (++" ") . reverse . words) str
    putStrLn reversed
